shop/src/main/java/ro/msg/learning/shop/controller/impl/OrderControllerImpl.java

```diff
@@ -0,0 +1,28 @@
+package ro.msg.learning.shop.controller.impl;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import ro.msg.learning.shop.controller.OrderController;
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.model.Order;
+import ro.msg.learning.shop.service.OrderService;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderControllerImpl implements OrderController {
+    private final OrderService orderService;
+
+    public OrderControllerImpl(OrderService orderService) {
+        this.orderService = orderService;
```

Comment:
Is this order service actually injected in here? You don't have an @Autowired or @RequiredArgsConstructor

Reply:
refactored the injection according to the guidelines using @RequiredArgsConstructor